### PR TITLE
Compile nco with ESMF

### PR DESCRIPTION
**Pull request by ocefpaf**
**Do not merge!** I am experiment nco with ESMF support. (It is still buggy and not advisable to do so :wink:) 

@bekozi you expertise is welcome here.

TODO:

- [ ] Add a re-grid test command

### CODE REVIEW

**Comment by bekozi** [comment]
@ocefpaf, are you looking for input on the test command?

**Comment by ocefpaf** [comment]
That too. I guess we need to come up with an use example. But @bekozi this is **very** low priority. Don't worry about it now.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,16 +19,18 @@ requirements:
         - hdf5
         - libnetcdf
         - udunits2
-        - krb5
         - expat
+        - krb5
+        - esmf
```
Comment (pelson) [inline recipe/meta.yaml]: We really need a nicer way of handling optional dependencies (nco still works without esmf even after this change, right?)

Comment (ocefpaf) [inline recipe/meta.yaml]: > We really need a nicer way of handling optional dependencies (nco still works without esmf even after this change, right?)

Indeed. The problem IMO happens in two places.

- conda does not have the concept of optional dependencies (like `pip install ipython[notebook]`);
- `nco` needs a special flag at build time to find ESMF.

We can get away with the former, but I cannot see how to circumvent the latter without building twice and renaming the package.

PS: Right now ESMF support is not working! This PR is just a test and I won't merge it.

**Comment by ocefpaf** [comment]
I will re-visit this once ESMF support in `nco` is more mature. I don't think it is worth right now.

**Comment by czender** [comment]
FYI NCO does not require ESMF at build-time. What I am talking about is the ESMF_RegridWeghtGen (ERWG) executable, not the library and headers. Although there is some #ifdef ESMF.,, code in NCO it is not ready for primetime so those flags should not be activated at build time.  We just want 'conda install nco' to simultaneously install ERWG in the conda environment. It has nothing to do with building NCO.

**Comment by ocefpaf** [comment]
Good to know. Thanks @czender! I will check if out `ESMF_RegridWeghtGen`  is built correctly and and I will it as a run dependency for `nco`.

**Comment by ocefpaf** [comment]
BTW: What would be an easy way to test it? Like a command that I can issue at test time to ensure `ERWG` was found and usable?

**Comment by czender** [comment]
There may not be an easy way to test ERWG. To test it in NCO I download some global datasets and regrid them. A simpler way would be to use ncks to generate two grids and then call ERWG to regrid them and make sure ERWG exits successfully. That would be three commands. If you want, I could supply three such commands. BTW, what I said above about ERWG also applies to TempestRemap. NCO (optionally) calls two Tempest executables: GenerateOverlapMesh and GenerateOfflineMap.

**Comment by ocefpaf** [comment]
>  That would be three commands. If you want, I could supply three such commands.

That would help. Thanks! (I can put that into a test_script that conda runs at test time. BTW [these](https://github.com/conda-forge/nco-feedstock/blob/master/recipe/meta.yaml#L38-L42) are the tests we run right now.)

> BTW, what I said above about ERWG also applies to TempestRemap. NCO (optionally) calls two Tempest executables: GenerateOverlapMesh and GenerateOfflineMap.

I will check is they are all there. I package ESMF but never got he time to really use it yet.

Ping @bekozi who might already know if we are good to go with those executables.

**Comment by czender** [comment]
OK, I suggest you use this as the test:
ESMF_RegridWeightGen --help
It is about as rigorous the NCO tests you run and will catch a bad install.

**Comment by bekozi** [comment]
We should be good with the ``make check`` in the ESMF build, but adding the the test like @czender suggested makes sure the paths are okay. So :+1:.

**Pull request closed without merging**